main/main.js

```diff
@@ -0,0 +1,171 @@
+const humidity = document.getElementById("humidity");
+const card = document.getElementById("card");
+const spinner = document.getElementById("spinner");
+
+const key = `82005d27a116c2880c8f0fcb866998a0`;
+
+const weatherTypes = [
+  "Clear",
+  "Clouds",
+  "Rain",
+  "Snow",
+  "Thunderstorm",
+  "Drizzle",
+  //7xx: Atmosphere there are many like Smoke, Mist, Haze, Fog etc will come in same category
+];
+
+const fetchData = async (lat, lon) => {
+  try {
+    const response = await fetch(
+      `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${key}`
+    );
+    if (response.status === 200) {
+      const data = await response.json();
+      const description = data.weather[0].description;
+      const tempInKelvin = data.main.temp;
+      const tempInCelcius = Math.floor(tempInKelvin - 273.15);
+      const cityName = data.name;
+      const countryName = data.sys.country;
+      const weatherType = data.weather[0].main;
+      const iconId = data.weather[0].icon;
+      const humid = data.main.humidity;
+      const wind = data.wind.speed;
+
+      return {
+        description,
+        tempInCelcius,
+        cityName,
+        countryName,
+        weatherType,
+        iconId,
+        humid,
+        wind,
+      };
+    } else {
+      console.log("ERROR! Wrong City Name");
+    }
+  } catch (e) {
+    console.log(e);
+  }
+};
+
+const fetchLocation = async (cityName) => {
+  try {
+    const response = await fetch(
+      `http://api.openweathermap.org/geo/1.0/direct?q=${cityName}&limit=5&appid=${key}`
+    );
+    const data = await response.json();
+    const { lat, lon } = data[0];
+    return { lat, lon };
+  } catch (e) {
+    console.log(e);
+    return { lat: "not found", lon: "not found" };
+  }
+};
+
+const getWeatherForCurrentLocation = async () => {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(async function (position) {
+      const latitude = position.coords.latitude;
+      const longitude = position.coords.longitude;
+      const weatherData = await fetchData(latitude, longitude);
+      displayWeatherData(weatherData);
+    });
+  } else {
+    console.log("Geolocation is not supported by this browser.");
+  }
+
+  navigator.geolocation.watchPosition(
+    function (position) {},
+    function (error) {
+      if (error.code == error.PERMISSION_DENIED) card.style.display = "block";
+      spinner.style.display = "none";
+    }
+  );
+};
+
+getWeatherForCurrentLocation();
+
+const getWeatherHandler = async () => {
+  if (input.value.trim() === "") {
+    console.log("Enter City Name");
+    displayWeatherData("Please Enter City Name");
+  } else {
+    const { lat, lon } = await fetchLocation(input.value);
+    const weatherData = await fetchData(lat, lon);
+    if (weatherData) {
+      weatherData.cityName = input.value;
+    }
+    displayWeatherData(weatherData);
+  }
+};
+
+const capitalizeFirstLetter = (word) => {
```

Comment:
without this is API not working? @aayushbatra02

Reply:
It is working. I am using this function to display city name in UI from input field